src/test/mir-opt/match_false_edges.rs

```diff
@@ -53,10 +53,11 @@ fn main() {
 //  bb0: {
 //      ...
 //      _2 = std::option::Option<i32>::Some(const 42i32,);
-//      _3 = discriminant(_2);
-//      _4 = &(promoted[1]: std::option::Option<i32>);
-//      _9 = discriminant(_2);
-//      switchInt(move _9) -> [0isize: bb5, 1isize: bb3, otherwise: bb7];
+//      ReadForMatch(_2);
+//      _7 = discriminant(_2);
+//      _9 = &(promoted[2]: std::option::Option<i32>);
```

Comment:
are these shallow borrows? can we have that show up somehow in the MIR?

Reply:
Sure, I'll change how they are printed